src/simperium/channel.js

```diff
@@ -39,6 +40,11 @@ internal.updateChangeVersion = function( cv ) {
  * @param {Object} change - the change to apply to the object
  */
 internal.changeObject = function( id, change ) {
+	// remove the change from our queue if the server already confirmed it
+	if ( this.localQueue.waitingFor( id ) ) {
```

Comment:
To explain a little bit more, the "ghost" of a given object is never written to until the change comes from the server. This applies to both changes originating from the same client or changes coming from other clients.

So if a client throws out changes it is waiting for like this it will never update the ghost with its changes that originate from itself.